server/utils/song.js

```diff
@@ -0,0 +1,67 @@
+var SpotifyWebApi = require('spotify-web-api-node');
+const { getAccessToken } = require('../../server/utils/playlist');
+const app = require('../server')
+
+  /**
+   * takes a track object and formats it
+   * @param {{}} track the track object
+   * @returns {{ id: String, name: String, artist: String, albumCover: String, duration: Number, uri: String }} formatted track
+   */
+  const formatSong = track => ({
+    name: track.name,
+    duration: track.duration_ms,
+    artist: track.album.artists[0].name,
+    uri:track.uri,
+    albumCover: track.album.images[0].url,
+    spotifyId:track.album.id
+  })
+
+function getSong(songUri, userID) {
+  return new Promise((resolve, reject) => {
+    
+    if (songUri == undefined) {
```

Comment:
What if songURI / userID  is an empty string or array? Make sure we're accounting for edge cases